somewherewarm/woocommerce-product-recommendations-custom-locations
# Update dev dependencies and fix broken build

## Description

This PR updates the dev dependencies and build scripts, allowing this project to be successfully built in the same dev environment as other extensions such as the base WooCommerce Product Recommendations.

Closes #9 

## Changes included

- `devDependencies` updated to be the same as the WooCommerce Product Recommendations repo
    - `node-sass` replaced with `sass`
    - `autoprefixer`'s `browsers` option changed to `overrideBrowserslist` to avoid deprecation notice
- `@somewherewarm/selectsw` dependency removed, as it is unused in this repo

## Testing instructions

On a system with **WooCommerce Product Recommendations** installed...

1. Clone repo
2. Run `npm i` - verify dependencies are installed without error (used to complain about missing `python2`)
3. Run `grunt build` - verify build completes without error
4. Activate plugin
5. Go to **WooCommerce** > **Recommendations** > **Custom Locations**
6. Add a custom location.
7. Edit the custom location and verify that the styling and functionality is correct... key points...
    - **Shortcode** meta box styling
    - **Copy shortcode** feature should work
8. Verify the following files were requested without error (if the above points in (7) are fine, this will be the case, but just to double-check!):
    - `assets/css/admin/wc-prl-cl-admin.css`
    - `assets/js/admin/wc-prl-cl-admin.js`

<img width="2396" alt="Screenshot 2024-08-26 at 16 08 00" src="https://github.com/user-attachments/assets/77e8ba9c-2763-4a54-95af-f65e0f0acbcc">
